### ISSUE TITLE
no-magic-numbers: false positive with bitshift

### ISSUE THREAD

**Issue by RomkeVdMeulen** [palantir/tslint#2059]
### Bug Report

- __TSLint version__: `4.3.1`
- __TypeScript version__: `2.1.5`
- __Running TSLint via__: gulp-tslint

#### TypeScript code being linted

```ts
export class MyClass {

	private static readonly MINIMAL_SPACE_BYTES = 2 << 20;

}
```

with `tslint.json` configuration:

```json
"no-magic-numbers": [true, -1, 0, 1, 2, 3, 4, 10, 100]
```

#### Actual behavior

(no-magic-numbers) my-class.ts[3, 53]: 'magic numbers' are not allowed

#### Expected behavior

No error

**Comment by andy-hanson** [comment]
This seems related to #1883. Although treating `2 <<` as not-magic makes more sense than allowing `60 *` as not-magic.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
